### ISSUE TITLE
Wall printing not optimized with "Optimize wall printing order" Cura version 5.4.0

### ISSUE THREAD

**Issue by kurolob** [Ultimaker/Cura#16165]
### Cura Version

5.4.0

### Operating System

Windows 10

### Printer

Custom

### Reproduction steps

Load a stl file that contains a cube with 3 vertical holes through the cube. In cura 5.4.0 and 4.10.0
Select "Optimize wall printing order" in every version
4.10.0 prints the inner wall and then outer wall for the first hole. Then inner wall and outer wall for the second hole. Then inner wall and outer wall for the third hole. Then inner wall and outer wall for the cube.
5.4.0 prints the inner wall for every wall. Then outer wall for every wall.
This can be a issue when printing a model with many islands for example.
![image](https://github.com/Ultimaker/Cura/assets/139664841/2c50d8bc-e042-4137-8d4f-f139be7fcaab)
![image](https://github.com/Ultimaker/Cura/assets/139664841/d3ca7a78-c20a-4de0-9ade-374871e4d0ea)

### Actual results

The wall are not optimized that may results in problems when printing "islands".

### Expected results

Should print like 4.10.0

### Add your .zip and screenshots here ⬇️

![image](https://github.com/Ultimaker/Cura/assets/139664841/2c50d8bc-e042-4137-8d4f-f139be7fcaab)
![image](https://github.com/Ultimaker/Cura/assets/139664841/d3ca7a78-c20a-4de0-9ade-374871e4d0ea)
[porta bateria com carregador e conversor-Body003.zip](https://github.com/Ultimaker/Cura/files/12063161/porta.bateria.com.carregador.e.conversor-Body003.zip)

**Comment by GregValiant** [comment]
Thanks for the report.
The Cura team will take a look.
It appears to me that it has something to do with the Wall Line Count.  Setting that to 3 produces a slightly different toolpath.  It also appears that the issue would be with the Inner-Walls.

**Comment by sityware** [comment]
I think I have the same issue with holes, on the bottom layer it causes so much back and forth I'm having unnecessary whisper strings.

It first draws the middle wall for each hole, then comes back to draw the inner hole of each, and then finally the outer wall for each.

![cura_hole_order](https://github.com/Ultimaker/Cura/assets/31103915/f42f2ac3-6cc8-4557-b435-583b1bf8fea4)

**Comment by GregValiant** [comment]
I get the same thing in my prints.  Rather than staying over one hole/feature and finishing it, Cura wants to do all the extrusions strictly by line type.  All the Inner-Walls, then all the Outer-Walls, then all the Extra Skin Walls, then the skin.  Certain combing modes along with monotonic can make the travel time pretty extensive.

**Comment by sityware** [comment]
I've created an example STL here that demonstrates, you can download here.
[https://mega.nz/file/8bxXTbIA#X2AQNHA6484EBXmXUEsFqz2OZB-vUeE5HVToRgEa4Io](url)

![image](https://github.com/Ultimaker/Cura/assets/31103915/fef324ee-a1f7-492d-8798-6927e999d543)

**Comment by sityware** [comment]
I have found a workaround, also thanks to @mottscanley

Group Outer Walls - Off
Extra Skin Wall Count - 0
Wall Line Count - Add +1

This now only goes over the holes once.

**Comment by Perds122** [comment]
Seems this issue has not been solved.
the setting of 'Group Outer Walls' is affecting 'Optimize wall printing order'
If 'GOW' is enabled 'OWPO'  has no affect

**Comment by Reynolds-Beta** [comment]
Cura 5.12, Windows 11.

As @Perds122 said, this issue is not resolved. Optimize Wall Printing Order does not seem to have a meaningful effect on the efficiency of the slice, and Group Outer Walls just makes it worse.

**Optimize Wall Printing Order**

Expected behavior:
All walls of a particular artifact in the model are printed continuously/sequentially for that particular artifact before moving onto the next. Example. A hole in the middle of a surface has all of its walls printed before moving onto the next.

Actual behavior:
All walls for a particular artifact in the model are not printed before moving onto the next item, only to come back at some later time to finish printing the rest of the walls. This increases travel and retractions on even the simplest model.

**Group Outer Walls**
This is an experimental feature which is ON by default when all experimental features should be OFF by default. This feature just makes the above behavior worse because it prints ALL outer walls no matter where they are, last. This further increases the travel and retractions. The fact that this experimental feature is ON by default is its own **bug**.

** Attachments **
I've included two different slices of the same relatively simple model. Starting from layer 2 (because layer one is reportedly not optimized) you can easily see that the walls are not printed all at once for each artifact in the model.

Final comment: There also an additional **bug** in that there are 3 walls printed for each artifact when the slicer is set for 2 (Wall thickness = 0.8mm, Wall line count = 2, as per the default for the Fast profile using a 0.4mm AA core).


[UMS5_DCM KX-12 Input Cup v5.zip](https://github.com/user-attachments/files/25948786/UMS5_DCM.KX-12.Input.Cup.v5.zip)

[Fast Profile.zip](https://github.com/user-attachments/files/25948859/Fast.Profile.zip)

**Comment by GregValiant** [comment]
We ask for project files because they contain your actual Cura settings and the model.  The "*.gcode" files and the "*.ufp" files can tell certain things, but not the whole story.

UM profiles (as developed for their machines) often have "Wall Ordering" set to "Outside to Inside" and when that is true, the "Outer Walls" are all printed first (Grouped) and then the machine will go back and print the inner walls.  That default was established during testing at UM on each of their printer models.

@Reynolds-Beta says:
_"There also an additional bug in that there are 3 walls printed for each artifact when the slicer is set for 2 (Wall thickness = 0.8mm, Wall line count = 2"_
There are 2 walls in this screenshot of the "ON" file.  What appears to be a third wall is the "Top/Bottom | Extra Skin Wall Count = 1".  That can be altered by the user (as can "Wall Ordering").

<img width="755" height="614" alt="Image" src="https://github.com/user-attachments/assets/3078e43f-0105-46bc-bf21-b16d6d3ec714" />

This report is from 2 1/2 years ago and refers to Cura 5.4.0.  In the intervening time between 5.4.0 and 5.12.0, things in Cura have changed.  In particular, I recall that this problem was worked on..  There could be a bug there, or the user's combination of settings isn't giving the expected results.

I'll leave this open, but if this problem is occurring with 5.12 then we could really use a separate bug report.
Either way - Please attach a project file that shows the problem that you are seeing.

**Comment by Reynolds-Beta** [comment]
@GregValiant

Fair enough. The only reason I posted to this bug is that it was still open - always the conundrum: New bug or continue with the original. 

Attached are two project files. For simplicity, I've disabled supports. Both are printed with the Balanced - Fast profile; One is default, the other is closer to what is expected.

A few items come to mind:

1. **Top/Bottom | Extra Skin Wall Count**: You're right - wasn't looking in Top/Bottom settings.

2. **Group Outer Walls**: Feature increases travel and really shouldn't be ON by default. This behavior leads the user to believe **Optimize Wall Printing Order** isn't being honored.

3. **Top/Bottom | Extra Skin Wall Count** (ON by default) / **Infill | Extra Infill Wall Count** (OFF by default): These features always print the extra walls as a group instead of as part of the continuous wall printing for each artifact in the model. Leads to more travel/retractions instead of being printed when the rest of the walls are printed. This behavior leads the user to believe **Optimize Wall Printing Order** isn't being honored.

[UMS5_DCM KX-12 Input Cup v5 - #2.zip](https://github.com/user-attachments/files/25971383/UMS5_DCM.KX-12.Input.Cup.v5.-.2.zip)

**Comment by GregValiant** [comment]
Thanks for the files.
_"1. **"Top/Bottom | Extra Skin Wall Count"**_
That's one of those settings that are hard to:  Find/Relate to/Remember to look at".  Having it in "Top Bottom" is fine as Cura relates it to Top/Bottom, but it's a wall setting too.  
_"2. **Group Outer Walls**"_  I found this really annoying and I have it turned off by default by adding an over-ride to my custom printer definition file (`"group_outer_walls": { "value": false }`).  It also seems to produce odd results when "Wall Ordering" is set to "inside-to-outside" which is the default for UM printers.  UM Testing must have shown that it produces better results on UM machines.  I find "Outside to Inside" to be troublesome with small holes and with overhangs.
_3. **"Extra Skin Wall Count"**_ and _**"Extra Infill Wall Count"**_
These do refer to two different things.  An extra wall around "Skins", and an extra wall around "Infill".  The Infill wall is never visible on a finished print but provides additional strength and opacity to the walls around infill.
_**"4. Optimize Wall Printing Order"**_
I know what the tooltip says, I just don't think this setting does much whether it is on or off.  That might be because I haven't found the combination of settings where it become useful.  It doesn't seem to hurt anything and I always leave it turned on.

There are 706 settings in Cura.  Most have been added in response to suggestions here, and by the work done by the Cura team in an effort to get the best results from UM printers.  Are there going to be cases where one setting interferes with another?  Of course.  If no one picks up the problem during beta testing, then the problem can make it into the stable version.  I'm not seeing any bug here though.  I'll leave the bug label on this for now and someone from the Cura team will take another look at this.

What they will want is a proper "Project File".  Use the "File | Save Project" command, then zip the 3mf file and post the zip folder here.  A project file is an exact duplicate of what you are working on and the way you have Cura set up to slice.  Your "ufp" files and the gcode file don't tell me what I need to know here.
